src/App.jsx

```diff
@@ -14,45 +16,90 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_ALPHABET = 'alphabetically';
+const SORT_LENGTH = 'length';
+
+const sortGoods = (goods, sortField, reverseField = false) => {
+  const currentGoods = [...goods];
+
+  if (sortField) {
+    currentGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_ALPHABET:
+          return good1.localeCompare(good2);
+        case SORT_LENGTH:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverseField) {
+    currentGoods.reverse();
+  }
+
+  return currentGoods;
+};
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [reverse, setReverse] = useState(false);
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const visibleGoods = sortGoods(goodsFromServer, sortField, reverse);
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          onClick={() => setSortField(SORT_ALPHABET)}
+          className={cn(
+            'button', 'is-info', { 'is-light': sortField !== SORT_ALPHABET },
+          )}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          onClick={() => setSortField(SORT_LENGTH)}
+          className={cn(
+            'button', 'is-success', { 'is-light': sortField !== SORT_LENGTH },
+          )}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          onClick={() => {
+            setReverse(!reverse);
+          }}
+          className={cn(
+            'button', 'is-warning', { 'is-light': !reverse },
+          )}
+        >
+          Reverse
+        </button>
+
+        {(sortField || reverse) && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => {
+              setSortField('');
+              setReverse(false);
+            }}
```

Comment:
it is bad practise to make anonymous arrow functions in jsx.
make handler in component and pass reference here

check other places